### PR TITLE
1493711: Fix --matches option for the list command.

### PR DESCRIPTION
**Pull request by awood**
Also fixes some underlying issues with the --pool-only option.

### CODE REVIEW

File: src/rhsmlib/services/entitlement.py [hunk 1]
```diff
@@ -78,14 +78,27 @@ def get_pools(self, pool_subsets=None, matches=None, pool_only=None, match_insta
             installed = products.InstalledProducts(self.cp).list(options['matches'])
             results['installed'] = [x._asdict() for x in installed]
         if 'consumed' in pool_subsets:
-            consumed = self.get_consumed_product_pools(**options)
-            results['consumed'] = [x._asdict() for x in consumed]
+            consumed = self.get_consumed_product_pools(service_level=service_level, matches=matches)
```
Comment (jirihnidek) [inline src/rhsmlib/services/entitlement.py]: Confirmed. It works:
```console
$ sudo busctl call com.redhat.RHSM1 /com/redhat/RHSM1/Entitlement com.redhat.RHSM1.Entitlement GetPools a{sv}a{sv} 2 pool_subsets as 1 "available" matches s "*workstation*" 0
```

File: src/rhsmlib/services/entitlement.py [hunk 2]
```diff
@@ -78,14 +78,27 @@ def get_pools(self, pool_subsets=None, matches=None, pool_only=None, match_insta
             installed = products.InstalledProducts(self.cp).list(options['matches'])
```
Comment (jirihnidek) [inline src/rhsmlib/services/entitlement.py]: I suggest to use same variable as for other cases ('consumed', 'available'):
```python
installed = products.InstalledProducts(self.cp).list(matches)
```

File: src/rhsmlib/services/entitlement.py [hunk 3]
```diff
@@ -78,14 +78,27 @@ def get_pools(self, pool_subsets=None, matches=None, pool_only=None, match_insta
...
+                service_level=service_level)
+            if pool_only:
+                results['available'] = [x['id'] for x in available]
+            else:
+                results['available'] = available
 
         return results
 
-    def get_consumed_product_pools(self, service_level=None, matches=None, **kwargs):
+    def get_consumed_product_pools(self, service_level=None, matches=None, pool_only=None):
```
Comment (jirihnidek) [inline src/rhsmlib/services/entitlement.py]: It seems that the `pool_only` is not used in this method and it is not called with this argument. It is possible to remove it.

**Comment by awood** [comment]
The test failure was caused by the issue fixed in #1707

**Comment by awood** [comment]
Bot, retest this please

**Comment by awood** [comment]
@jirihnidek Mind re-reviewing?  I think this is ready to merge.

**Pull request merged**